SmartTutor/ContentModel/LearningObjects/ChallengeModel/FulfillmentStrategy/ChallengeFulfillmentStrategy.cs

```diff
@@ -0,0 +1,10 @@
+﻿using RepositoryCompiler.CodeModel.CaDETModel.CodeItems;
+using System.Collections.Generic;
+
+namespace SmartTutor.ContentModel.LearningObjects.ChallengeModel.FulfillmentStrategy
+{
+    public abstract class ChallengeFulfillmentStrategy
+    {
+        public abstract bool CheckChallengeFulfillment(List<CaDETClass> submittetClasses, Challenge challenge);
```

Comment:
We can remove the challenge, and I would rename this list of classes (that is passed around in several places) to solutionAttempt or challengeSolutionAttempt.

submittedClasses is a bit redundant - first it is clear they are classes from the type, and submitted is half-informative since this data arrives on an endpoint through a POST request, making it submitted.